mypy/build.py

```diff
@@ -275,9 +275,21 @@ def compute_search_paths(sources: List[BuildSource],
     if alt_lib_path:
         mypypath.insert(0, alt_lib_path)
 
+    package_path = tuple(_get_site_packages_dirs(options.python_executable))
+    for site_dir in package_path:
+        assert site_dir not in lib_path
+        if site_dir in mypypath:
```

Comment:
There are of course many variant spellings of the same path that won't be caught by this check. I don't think it's necessary to worry about that (normalizing the paths feels kind of expensive), but we shouldn't feel *too* complacent. :-)

Reply:
Yeah, it won't be perfect, but I think it will catch and help debug some more common cases, as well as avoid crashes in those circumstances.